src/background/adblocker.js

```diff
@@ -94,6 +95,37 @@ if (__PLATFORM__ === 'firefox') {
   });
 }
 
+let ENABLE_CHROMIUM_SCRIPTING_ONRESPONSESTARTED = false;
+if (__PLATFORM__ === 'chromium') {
+  store.resolve(Config).then((config) => {
+    const enabled = config.hasFlag(FLAG_CHROMIUM_SCRIPTING_ONRESPONSESTARTED);
+    if (!enabled) contentScripts.unregisterAll();
+
+    ENABLE_CHROMIUM_SCRIPTING_ONRESPONSESTARTED = enabled;
+  });
+}
+
+// Use `documentId` field from Chromium `webRequest` and `webNavigation` events
+// to determine if cosmetic injection is already happened. Note that this is
+// not available on Firefox yet. If we need to listen from multiple events on
+// Firefox, we need to calculate a hash of tab and frame to generate `documentId`
+// on the fly.
+/**
+ * @type {Set<string>}
+ */
+const executedDocumentIds = new Set();
```

Comment:
This may not work. Does `executeScript` guarantee the scriptlet execution? What if document is not created yet?

I think we should try to inject many times, but be sure that the scriptlet is run only once. That may require so flag in the execution environment. But such flag should not be detectable by a page script as it could be used to detect the adblocker.

Reply:
`executeScript` doesn't guarantee the script execution in the page. Also, the timing never matched by running `executeScript` continuously by using loop until they actually respond with results. I believe this is by the design of Chromium engine. Even they doesn't guarantee some time related function correctly.

In the given events, the documents are already all created. From the main frame type of on response started event, the document was always available. Also, navigation on committed event was not touched.